### PR TITLE
datepicker: create injectable for date formats and bundle it along with date adapter into MdNativeDateModule

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/lib/datepicker/calendar.ts [hunk 1]
```diff
@@ -95,7 +107,8 @@ export class MdCalendar<D> implements AfterContentInit {
   /** The label for the current calendar view. */
   get _periodButtonText(): string {
     return this._monthView ?
-        this._dateAdapter.getMonthYearName(this._activeDate, 'short').toLocaleUpperCase() :
+        this._dateAdapter.format(this._activeDate, this._dateFormats.display.monthYearLabel)
+            .toLocaleUpperCase() :
```
Comment (jelbourn) [inline src/lib/datepicker/calendar.ts]: Shouldn't the upper-casing be part of what the formatter does?

Comment (mmalerba) [inline src/lib/datepicker/calendar.ts]: I don't think either the format language used by Intl.DatetimeFormat or Moment.js gives control over the case, so we're stuck doing this

**Comment by fxck** [comment]
@mmalerba anything else missing before merging into master?

**Comment by mmalerba** [comment]
@fxck This was the last major feature, all I have left to do is make sure it works well on different browsers and screen readers.

**Comment by fxck** [comment]
Ok, cool. I tried c&p it into my project and fixing the paths meanwhile, looks like there are some problems with it not being rebased as well.

**Comment by mmalerba** [comment]
I just rebased it, give it another try

**Comment by fxck** [comment]
It wasn't the problem in the end, underlineRef currently not existing on input container on current master was. I've changed it to elementRef instead for now.

**Comment by fxck** [comment]
@mmalerba do you have any screenshot of how datetimepicker will look?

**Pull request merged**